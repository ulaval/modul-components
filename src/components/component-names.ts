export const ACCORDION_GROUP_NAME: string = 'm-accordion-group';
export const ACCORDION_NAME: string = 'm-accordion';
export const LIST_ITEM_NAME: string = 'm-list-item';
export const BUTTON_NAME: string = 'm-button';
export const BUTTON_GROUP_NAME: string = 'm-button-group';
export const CAROUSEL_NAME: string = 'm-carousel';
export const CAROUSEL_ITEM_NAME: string = 'm-carousel-item';
export const CHECKBOX_NAME: string = 'm-checkbox';
export const DATEFIELDS_NAME: string = 'm-datefields';
export const DATEPICKER_NAME: string = 'm-datepicker';
export const DIALOG_NAME: string = 'm-dialog';
export const DROPDOWN_NAME: string = 'm-dropdown';
export const DROPDOWN_ITEM_NAME: string = 'm-dropdown-item';
export const DROPDOWN_GROUP_NAME: string = 'm-dropdown-group';
export const DYNAMIC_TEMPLATE_NAME: string = 'm-dynamic-template';
export const FILE_SELECT_NAME: string = 'm-file-select';
export const FILE_UPLOAD_NAME: string = 'm-file-upload';
export const EDIT_WINDOW_NAME: string = 'm-edit-window';
export const FLEX_TEMPLATE_NAME: string = 'm-flex-template';
export const I18N_NAME: string = 'm-i18n';
export const ICON_NAME: string = 'm-icon';
export const ICON_BUTTON_NAME: string = 'm-icon-button';
export const INPUT_STYLE_NAME: string = 'm-input-style';
export const LIMIT_TEXT_NAME: string = 'm-limit-text';
export const LINK_NAME: string = 'm-link';
export const MESSAGE_NAME: string = 'm-message';
export const MODAL_NAME: string = 'm-modal';
export const NAVBAR_NAME: string = 'm-navbar';
export const NAVBAR_ITEM_NAME: string = 'm-navbar-item';
export const MENU_NAME: string = 'm-menu';
export const MENU_ITEM_NAME: string = 'm-menu-item';
export const PANEL_NAME: string = 'm-panel';
export const PHONE_NUMBER_NAME: string = 'm-phone-number';
export const POPPER_NAME: string = 'm-popper';
export const POPUP_NAME: string = 'm-popup';
export const PROGRESS_NAME: string = 'm-progress';
export const RADIO_NAME: string = 'm-radio';
export const RADIO_GROUP_NAME: string = 'm-radio-group';
export const RADIO_STYLE_NAME: string = 'm-radio-style';
export const SCROLL_TOP_NAME: string = 'm-scroll-top';
export const SIDEBAR_NAME: string = 'm-sidebar';
export const SLIDER_NAME: string = 'm-slider';
export const SPINNER_NAME: string = 'm-spinner';
export const STATUS_NAME: string = 'm-status';
export const STEP_NAME: string = 'm-step';
export const STEPPERS_NAME: string = 'm-steppers';
export const STEPPERS_ITEM_NAME: string = 'm-steppers-item';
export const SWITCH_NAME: string = 'm-switch';
export const TAB_PANEL_NAME: string = 'm-tab-panel';
export const TABS_NAME: string = 'm-tabs';
export const TEMPLATE_NAME: string = 'm-template';
export const TEXTAREA_NAME: string = 'm-textarea';
export const TEXTFIELD_NAME: string = 'm-textfield';
export const TIMEPICKER_NAME: string = 'm-timepicker';
export const TOOLTIP_NAME: string = 'm-tooltip';
export const UPLOAD_NAME: string = 'm-upload';
export const UPLOAD_INPUT_NAME: string = 'm-upload-input';
export const UPLOAD_DRAGDROP_NAME: string = 'm-upload-dragdrop';
export const UPLOAD_FILESLIST_NAME: string = 'm-upload-fileslist';
export const VALIDATION_MESSAGE_NAME: string = 'm-validation-message';
