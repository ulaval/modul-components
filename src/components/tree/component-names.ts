export const TREE_NODE_NAME: string = 'm-tree-node';
export const TREE_ICON_NAME: string = 'm-tree-icon';
