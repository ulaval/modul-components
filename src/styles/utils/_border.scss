.m-u--app-body {
    .m-u {
        &--border {
            border: $m-border-width solid;

            &--l {
                border: $m-border-width--l solid;
            }

            &--s {
                border: $m-border-width--s solid;
            }

            &-top {
                border-top: $m-border-width solid;

                &--l {
                    border-top: $m-border-width--l solid;
                }

                &--s {
                    border-top: $m-border-width--s solid;
                }
            }

            &-right {
                border-right: $m-border-width solid;

                &--l {
                    border-right: $m-border-width--l solid;
                }

                &--s {
                    border-right: $m-border-width--s solid;
                }
            }

            &-bottom {
                border-bottom: $m-border-width solid;

                &--l {
                    border-bottom: $m-border-width--l solid;
                }

                &--s {
                    border-bottom: $m-border-width--s solid;
                }
            }

            &-left {
                border-left: $m-border-width solid;

                &--l {
                    border-left: $m-border-width--l solid;
                }

                &--s {
                    border-left: $m-border-width--s solid;
                }
            }

            &-top-bottom {
                border-top: $m-border-width solid;
                border-bottom: $m-border-width solid;

                &--l {
                    border-top: $m-border-width--l solid;
                    border-bottom: $m-border-width--l solid;
                }

                &--s {
                    border-top: $m-border-width--s solid;
                    border-bottom: $m-border-width--s solid;
                }
            }

            &-right-left {
                border-right: $m-border-width solid;
                border-left: $m-border-width solid;

                &--l {
                    border-right: $m-border-width--l solid;
                    border-left: $m-border-width--l solid;
                }

                &--s {
                    border-right: $m-border-width--s solid;
                    border-left: $m-border-width--s solid;
                }
            }
        }

        &--no-border {
            border: none;

            &-top {
                border-top: none;
            }

            &-right {
                border-right: none;
            }

            &-bottom {
                border-bottom: none;
            }

            &-left {
                border-left: none;
            }

            &-top-bottom {
                border-top: none;
                border-bottom: none;
            }

            &-right-left {
                border-right: none;
                border-left: none;
            }
        }
    }
}
