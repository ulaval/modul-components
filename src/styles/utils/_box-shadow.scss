.m-u--app-body {
    .m-u--overflow {
        &,
        &--auto,
        &--scroll,
        &--visible {
            @include m-scrollbar();
        }

        &--auto {
            overflow: auto;
        }

        &--scroll {
            overflow: scroll;
        }

        &--visible {
            overflow: visible;
        }

        &--hidden {
            overflow: hidden;
        }
    }
}
