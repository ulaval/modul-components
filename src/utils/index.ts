import Vue, { PluginObject } from 'vue';

import MediaQueriesPlugin from './media-queries/media-queries';
import ModulPlugin from './modul/modul';
import MessagesPlugin from './i18n/i18n';
import HttpPlugin, { HttpPluginOptions } from './http/http';
import SpritesPlugin from './svg/sprites';
import * as TouchPlugin from 'vue-touch';
import ConfirmPlugin from './modal/confirm';
import AlertPlugin from './modal/alert';
import FilePlugin from '../utils/file/file';
import { PortalPluginInstall } from 'portal-vue';
import LoggerPlugin, { ConsoleOptions } from './logger/logger';

export interface UtilsPluginOptions {
    httpPluginOptions?: HttpPluginOptions;
    consoleOptions?: ConsoleOptions;
}

const UtilsPlugin: PluginObject<any> = {
    install(v, options): void {
        if (!v.prototype.$log) {
            Vue.use(LoggerPlugin, options ? options.consoleOptions : undefined);
        } else if (options) {
            v.prototype.$log.setConsoleOptions(options.consoleOptions);
        }

        Vue.use(MediaQueriesPlugin);
        Vue.use(ModulPlugin);
        Vue.use(MessagesPlugin);
        Vue.use(HttpPlugin, options ? options.httpPluginOptions : undefined);
        Vue.use({ install: PortalPluginInstall });
        Vue.use(SpritesPlugin);
        Vue.use(TouchPlugin, { name: 'v-touch' });
        Vue.use(ConfirmPlugin);
        Vue.use(AlertPlugin);
        Vue.use(FilePlugin);
    }
};

export default UtilsPlugin;
