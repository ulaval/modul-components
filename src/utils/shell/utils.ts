import Vue from 'vue';
import { Shell, PackageOptions } from '@ulaval/shell-ui/dist/shell/shell';

declare var __webpack_public_path__: string | undefined;

export type CreerInstanceVueFn = (shell: Shell, rootPath: string) => Vue;

export function setupShell(nomPackage: string, creerInstanceVue: CreerInstanceVueFn): void {
    window[nomPackage] = {
        mount(shell: Shell, options: PackageOptions): Promise<void> {
            if (this.vue) {
                console.warn('Already mounted.');
                return Promise.resolve();
            }

            __webpack_public_path__ = options.repoPublicPath;

            this.idElementRacine = options.rootElement;

            let rootPath: string = options.rootPath as string;
            if (rootPath[rootPath.length - 1] !== '/') {
                rootPath += '/';
            }

            this.vue = creerInstanceVue(shell, rootPath);

            this.vue.$mount('#' + this.idElementRacine);

            return Promise.resolve();
        },

        unmount(): Promise<void> {
            if (!this.vue) {
                return Promise.resolve();
            }

            const el = this.vue.$el;
            this.vue.$destroy();
            this.vue = undefined;

            let div = document.createElement('div');
            div.id = this.idElementRacine;
            if (el.parentNode) {
                el.parentNode.replaceChild(div, el);
            }

            return Promise.resolve();
        }
    };
}
