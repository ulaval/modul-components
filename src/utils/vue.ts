import Vue from 'vue';
import { Messages } from './i18n';
import { AxiosStatic } from 'axios';

export class ModulVue extends Vue {
    public $i18n: Messages;
    public $http: AxiosStatic;
}
