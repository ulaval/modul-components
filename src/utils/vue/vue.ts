import Vue from 'vue';
import Component from 'vue-class-component';
import { Messages } from '../i18n/i18n';
import { HttpService } from '../http/http';
import { FileService } from '../file/file';
import { MediaQueries } from '../media-queries/media-queries';
import { Modul } from '../modul/modul';
import { ConfirmFunction } from '../modal/confirm';
import { AlertFunction } from '../modal/alert';

console.debug('TODO: explore usage of TS declare syntax');
// declare module 'vue/types/vue' {
//     interface Vue {
//         $i18n: Messages;
//     }
// }

@Component
export class ModulVue extends Vue {
    public $i18n: Messages;
    public $http: HttpService;
    public $mq: MediaQueries;
    public $modul: Modul;
    public $confirm: ConfirmFunction;
    public $alert: AlertFunction;
    public $file: FileService;

    protected getParent<T extends Vue>(test: (obj: Vue) => boolean): T | undefined {
        let p: Vue = this.$parent;
        while (p && !test(p)) {
            p = p.$parent;
        }
        return p as any;
    }

    protected as<T>(): T {
        let result: any = this;
        return result as T;
    }
}
