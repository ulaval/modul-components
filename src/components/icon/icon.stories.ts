import { object, select, text } from '@storybook/addon-knobs';
import { storiesOf } from '@storybook/vue';
import Vue from 'vue';
import { componentsHierarchyRootSeparator } from '../../../conf/storybook/utils';
import { ICON_NAME } from '../component-names';
import IconPlugin from './icon';

Vue.use(IconPlugin);



const ICONLIST_SHORTHAND: {} = {
    'profile': 'profile',
    'information': 'information',
    'confirmation': 'confirmation',
    'warning': 'warning',
    'error': 'error',
    'calendar': 'calendar',
    'hint': 'hint',
    'clock': 'clock'
};

const ICONLIST_LONGHAND: {} = {
    'm-svg__profile': 'm-svg__profile',
    'm-svg__information': 'm-svg__information',
    'm-svg__confirmation': 'm-svg__confirmation',
    'm-svg__warning': 'm-svg__warning',
    'm-svg__error': 'm-svg__error',
    'm-svg__calendar': 'm-svg__calendar',
    'm-svg__hint': 'm-svg__hint',
    'm-svg__clock': 'm-svg__clock'
};

storiesOf(`${componentsHierarchyRootSeparator}${ICON_NAME}`, module)


    .add('name (short)', () => ({
        props: {
            name: {
                default: select('name', ICONLIST_SHORTHAND, 'profile')
            }
        },
        template: `<div>
                       <m-icon :name="name"></m-icon>
                       <span style="margin-left: 15px">Current name: <span style="color:red">{{ name }}</span></span>
                   </div>`
    }))
    .add('name (long)', () => ({
        props: {
            name: {
                default: select('name', ICONLIST_LONGHAND, 'm-svg__profile')
            }
        },
        template: `<div>
                       <m-icon :name="name"></m-icon>
                       <span style="margin-left: 15px">Current name: <span style="color:red">{{ name }}</span></span>
                   </div>`
    }))
    .add('svgTitle', () => ({
        props: {
            svgTitle: {
                default: text('svgTitle', 'Title1')
            }
        },
        template: `<div>
                       <m-icon :svgTitle="svgTitle" name="profile"></m-icon>
                       <span style="margin-left: 15px">Current svgTitle: <span style="color:red">{{ svgTitle }}</span></span>
                   </div>`
    }))
    .add('size', () => ({
        props: {
            size: {
                default: text('size (px)', '30')
            }
        },
        template: `<div>
                       <m-icon :size="size" name="profile"></m-icon>
                       <span style="margin-left: 15px">Current icon size: <span style="color:red">{{ size }}</span></span>
                   </div>`
    }))
    .add('showNameAsClass', () => ({
        props: {
            name: {
                default: select('icon name as class', ICONLIST_SHORTHAND, 'profile')
            }
        },
        template: `<div>
                       <m-icon :name="name" :showNameAsClass="true"></m-icon>
                       <span style="margin-left: 15px">Current className on svg Icon: <span style="color:red">{{ name }}</span></span>
                   </div>`
    }));

storiesOf(`${componentsHierarchyRootSeparator}${ICON_NAME}/badge`, module)


    .add('state', () => ({
        props: {
            badge: {
                default: object("{ state: 'completed|error|warning' }", { state: 'completed' })
            },
            size: {
                default: text('size (px)', '30')
            }
        },
        template: '<m-icon :size="size" v-m-badge="badge" name="profile"></m-icon>'
    }))
    .add('offsetX', () => ({
        props: {
            badge: {
                default: object("{ state: 'completed|error|warning', offsetX: 'valueX' }", {
                    state: 'completed',
                    offsetX: '9px'
                })
            }
        },
        template: '<m-icon size="30px" v-m-badge="badge" name="profile"></m-icon>'
    }))
    .add('offsetY', () => ({
        props: {
            badge: {
                default: object("{ state: 'completed|error|warning', offsetY: 'valueY' }", {
                    state: 'completed',
                    offsetY: '7px'
                })
            }
        },
        template: '<m-icon size="30px" v-m-badge="badge" name="profile"></m-icon>'
    }))
    .add('state, offsetY, offsetX', () => ({
        props: {
            badge: {
                default: object("{ state: 'completed|error|warning', offsetX: 'valueX', offsetY: 'valueY' }",
                    { state: 'completed', offsetX: '-5px', offsetY: '-8px' })
            },
            size: {
                default: text('size (px)', '30')
            }
        },
        template: '<m-icon :size="size" v-m-badge="badge" name="profile"></m-icon>'
    }));
