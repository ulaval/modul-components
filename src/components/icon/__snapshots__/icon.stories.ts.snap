// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots components|m-icon name (long) 1`] = `
<div>
  <svg
    aria-hidden="true"
    class="m-icon"
    height="20px"
    width="20px"
  >
    <!---->
     
    <use
      aria-hidden="true"
      class=""
    />
  </svg>
   
  <span
    style="margin-left: 15px;"
  >
    Current name: 
    <span
      style="color: red;"
    >
      m-svg__profile
    </span>
  </span>
</div>
`;

exports[`Storyshots components|m-icon name (short) 1`] = `
<div>
  <svg
    aria-hidden="true"
    class="m-icon"
    height="20px"
    width="20px"
  >
    <!---->
     
    <use
      aria-hidden="true"
      class=""
    />
  </svg>
   
  <span
    style="margin-left: 15px;"
  >
    Current name: 
    <span
      style="color: red;"
    >
      profile
    </span>
  </span>
</div>
`;

exports[`Storyshots components|m-icon showNameAsClass 1`] = `
<div>
  <svg
    aria-hidden="true"
    class="m-icon"
    height="20px"
    width="20px"
  >
    <!---->
     
    <use
      aria-hidden="true"
      class="profile"
    />
  </svg>
   
  <span
    style="margin-left: 15px;"
  >
    Current className on svg Icon: 
    <span
      style="color: red;"
    >
      profile
    </span>
  </span>
</div>
`;

exports[`Storyshots components|m-icon size 1`] = `
<div>
  <svg
    aria-hidden="true"
    class="m-icon"
    height="30"
    width="30"
  >
    <!---->
     
    <use
      aria-hidden="true"
      class=""
    />
  </svg>
   
  <span
    style="margin-left: 15px;"
  >
    Current icon size: 
    <span
      style="color: red;"
    >
      30
    </span>
  </span>
</div>
`;

exports[`Storyshots components|m-icon svgTitle 1`] = `
<div>
  <svg
    class="m-icon"
    height="20px"
    width="20px"
  >
    <title>
      Title1
    </title>
     
    <use
      aria-hidden="true"
      class=""
    />
  </svg>
   
  <span
    style="margin-left: 15px;"
  >
    Current svgTitle: 
    <span
      style="color: red;"
    >
      Title1
    </span>
  </span>
</div>
`;

exports[`Storyshots components|m-icon/badge offsetX 1`] = `
<svg
  aria-hidden="true"
  class="m-icon"
  height="30px"
  style="overflow: visible;"
  width="30px"
>
  <!---->
   
  <use
    aria-hidden="true"
    class=""
  />
</svg>
`;

exports[`Storyshots components|m-icon/badge offsetY 1`] = `
<svg
  aria-hidden="true"
  class="m-icon"
  height="30px"
  style="overflow: visible;"
  width="30px"
>
  <!---->
   
  <use
    aria-hidden="true"
    class=""
  />
</svg>
`;

exports[`Storyshots components|m-icon/badge state 1`] = `
<svg
  aria-hidden="true"
  class="m-icon"
  height="30"
  style="overflow: visible;"
  width="30"
>
  <!---->
   
  <use
    aria-hidden="true"
    class=""
  />
</svg>
`;

exports[`Storyshots components|m-icon/badge state, offsetY, offsetX 1`] = `
<svg
  aria-hidden="true"
  class="m-icon"
  height="30"
  style="overflow: visible;"
  width="30"
>
  <!---->
   
  <use
    aria-hidden="true"
    class=""
  />
</svg>
`;
